Google.Api.Generator/CodeGenerator.cs

```diff
@@ -215,6 +219,9 @@ internal static class CodeGenerator
                 var unitTestsCsprojContent = CsProjGenerator.GenerateUnitTests(ns);
                 var unitTestsCsprojFilename = $"{unitTestsPathPrefix}{ns}.Tests.csproj";
                 yield return new ResultFile(unitTestsCsprojFilename, unitTestsCsprojContent);
+                // Generate gapic_metadata.json
+                var gapicMetadataJsonContent = MetadataGenerator.GenerateGapicMetadataJson(allServiceDetails);
```

Comment:
FYI we'll also need to guard this behind a CLI flag (please see "File Generation" in the spec). Up to you whether to put that in this one or a separate PR.

Reply:
It'll have to be a separate PR

Reply:
https://github.com/googleapis/gapic-generator-csharp/issues/316 to track